onnxruntime/test/providers/cpu/math/matmul_test.cc

```diff
@@ -211,8 +208,19 @@ TEST(MathOpTest, MatMulFloatType) {
   RunMatMulTest<float>(7, false, false);
 }
 
-// To Test XNNPACK, Matrix B must be constant
-TEST(MathOpTest, MatMulFloatType_ConstantB) {
+TEST(MathOpTest, MatMulFloat16) {
+  // TODO: Unskip when fixed #41968513
+  if (DefaultDmlExecutionProvider().get() != nullptr) {
+    GTEST_SKIP() << "Skipping because of the following error: Assertion failed: m_bufferTensorDesc.TotalTensorSizeInBytes >= ComputeByteSizeFromDimensions(nonBroadcastDimensions, dataType)";
+  }
+  RunMatMulTest<MLFloat16>(14, false, false);
+}
+
+TEST(MathOpTest, MatMulDoubleType) {
+  RunMatMulTest<double>(7);
+}
+
+TEST(MathOpTest, MatMulFloatTypeInitializer) {
```

Comment:
nit: Can we keep the pair of tests (MuatMulFloat/MatMulFloatTypeInitializer) for a data type adjacent so it's easier to see their relationship? Or just run both variants in the one test (we do that in lots of other places).